### PR TITLE
Components: Fix responsive across elements with box-sizing removal

### LINKED ISSUE #28485: My Jetpack: fix responsiveness issue with Jetpack dashboard cards
Reported on p1HpG7-kcV-p2

Caused by a change introduced between Jetpack 11.7.a.7 and Jetpack 11.7.a.9

Changelog for 11.7.a.9 [here](https://plugins.trac.wordpress.org/browser/jetpack/tags/11.7-a.9/CHANGELOG.md) and [here](https://github.com/Automattic/jetpack-production/releases/tag/11.7-a.9)

Before the bug was introduced, this is the expected view (11.7.a.7 and previous versions):

Image

After the bug was introduced, this is the current view (from 11.7.a.9 on):

Image

### PR DESCRIPTION
**Pull request by renatoagds**
As we removed `box-sizing: border-box` from `base-styles` ( #28024 ), all elements got moved to `content-box` (the default rule) since we relied on `base-styles` for applying `border-box`. This rule (`content-box`) calculates the element size differently (not including borders/margins/padding). This movement completely broke our responsiveness across the pages.

This was happening in `My Jetpack`, `VideoPress` and all pages that uses our `Section/Layout` components.

Closes: #28485 

## Proposed changes:

* Remove local `box-sizing` from `My Jetpack`.
* Add `box-sizing: border-box` on `ThemeProvider` with support to include more global rules in the future.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

p1HpG7-kcV-p2

## Testing instructions:

* Build `My Jetpack` and `VideoPress` or other plugin that uses `Section/Layout` component.
* Open it initial page.
* Reduce the page `width` manually.
* You should see that `padding` and/or `margins` are not outside the width count.
* Responsive should work normally.

### CODE REVIEW

File: projects/js-packages/components/components/theme-provider/globals.module.scss [hunk 1]
```diff
@@ -0,0 +1,6 @@
+/* DO NOT USE THIS TO OVERRIDE STYLES */
+.theme {
```
Comment (retrofox) [inline projects/js-packages/components/components/theme-provider/globals.module.scss]: A minor one, I think the `.theme` is confusing.  Maybe something aligned with the prop name `withGlobalStyles`?

Comment (renatoagds) [reply projects/js-packages/components/components/theme-provider/globals.module.scss]: `theme` was related to `theme-provider` but we could change it, no problem.

Comment (renatoagds) [reply projects/js-packages/components/components/theme-provider/globals.module.scss]: done: https://github.com/Automattic/jetpack/pull/28489/commits/e159e83bdf1b569d5f9f5a2389454daa9d3de4ea

**Comment by lhkowalski** [comment]
Tested the fix. Before it, I was seeing the page this way:

Screen Shot 2023-01-20 at 10 27 36

After the fix, I see the page this way:

Screen Shot 2023-01-20 at 10 42 49

Screen Shot 2023-01-20 at 10 43 03

The fix works as expected. Well done!

**Review by lhkowalski — APPROVED** [review]
Code change looks good.

**Pull request merged**